test/test_train_imagenet.py

```diff
@@ -150,9 +165,17 @@ def train_loop_fn(model, loader, device, context):
             lr=FLAGS.lr,
             momentum=FLAGS.momentum,
             weight_decay=5e-4))
+
+    scheduler = get_scheduler_for_model(
+        FLAGS.model, optimizer, num_training_steps_per_epoch)
     tracker = xm.RateTracker()
     model.train()
     for x, (data, target) in loader:
+      # TODO BEFORE SUBMIT(zcain): Should we call scheduler.step() before or
+      # after optimizer.step() ?
+      if scheduler:
+        scheduler.step(epoch, x, summary_writer=writer if should_report_lr(
```

Comment:
making summary_writer a constructor arg would simplify here too.

Reply:
Good call. The scheduler.step() call is much simpler now